### ISSUE TITLE
common configuration of husky, eslint and prettier across all packages

### ISSUE THREAD

**Issue by Nishchit14** [firecamp-dev/firecamp#85]
**Is your feature request related to a problem? Please describe.**
Currently, the code formatting is implemented package-wise as the majority of packages were independent repos before the OSS launch. So there is no code formatting, prettify,i and commit standards are defined for the whole app.

**Describe the solution you'd like**
There should be a common package like `@firecamp/eslint`, or `@firecamp/prettify` available. we can install them in the mono repo to maintain the standards and reuse the same configuration across all packages.

**Comment by nagy-nabil** [comment]
i can give this issue a try, if it's still available

**Comment by Nishchit14** [comment]
@nagy-nabil yes, you can share the PR.

**Comment by bathini79** [comment]
So, As Far As I understand . We're a expectin a repo with common configurations in packages folder. ANd Inject this with dependencies in mono repos?

**Comment by Nishchit14** [comment]
yes, you're right @bathini79

**Comment by bathini79** [comment]
Thanks for the conformation! I started working on it

**Comment by Nishchit14** [comment]
> Thanks for the conformation! I started working on it

Great, eager to see it in action. I am more excited about this issue. @bathini79

**Comment by bathini79** [comment]
I have a follow up question now,  If we could simply configure this at root level . And then it would be applicable accross all the mono repos , then what is the need of this kind of configuration?

**Comment by Nishchit14** [comment]
@bathini79 we need to create an independent package for **code quality standards**. so we can reuse it everywhere (backend, frontend, plugins, npm packages, oss/non-oss packages). we need to prepare a package once and then reuse it anywhere.

this package would be exporting
1. eslint config
2. prettier config
3. anything else?

**Comment by bathini79** [comment]
oh! That makes so much sense now, Thanks !

**Comment by bathini79** [comment]
@Nishchit14  This is my first working with this kind of setup. Do I need Push the code for just the lint -manager and get a repo created and then i can install where ever we want?

**Comment by Nishchit14** [comment]
I am happy that you're showing good courage to work on this issue being a first contributor. Let me bring some help for you.

We can create the following new packages (under `packages` folder at root)
- [ ] 1. `@firecamp/eslint`     (folder name `firecamp-eslint`)
- [ ] 2. `@firecamp/prettier` (folder name `firecamp-prettier`)
- [ ] 3. `@firecamp/tsconfig`  (folder name `firecamp-tsconfig`)

Here are some reference packages from other friend oss projects

1. https://github.com/calcom/cal.com/tree/main/packages/tsconfig
2. https://github.com/calcom/cal.com/tree/main/packages/config
3. https://github.com/calcom/cal.com/tree/main/packages/eslint-plugin
4. https://github.com/documenso/documenso/tree/feat/refresh/packages/eslint-config
5. https://github.com/documenso/documenso/tree/feat/refresh/packages/tsconfig

I see @dhoni77 has contributed in cal.com, if you can guide or correct us here then it would be a great help @dhoni77

@bathini79 you can start with creating the first package `@firecamp/eslint`, if you have any questions or need help then feel free to ask anytime here or in discord.

**Comment by bathini79** [comment]
Hey @Nishchit14 I'm good with creating the files and setting up the config. But how do I push these changes that's thing I'm wondering. I've tried with git submodule . But it's not working as expected . It's just saying that it is a submodule and does not show any code in  the git

**Comment by Dhoni77** [comment]
> I am happy that you're showing good courage to work on this issue being a first-time contributor. Let me bring some help for you.
> 
> We can create the following new packages (under the `packages` folder at the root)
> 
> * [ ]  1. `@firecamp/eslint`     (folder name `firecamp-eslint`)
> * [ ]  2. `@firecamp/prettier` (folder name `firecamp-prettier`)
> * [ ]  3. `@firecamp/tsconfig`  (folder name `firecamp-tsconfig`)
> 
> > Later we'll figure out the place to keep those config packages separately
> 
> Here are some reference packages from other friend oss projects
> 
> 1. https://github.com/calcom/cal.com/tree/main/packages/tsconfig
> 2. https://github.com/calcom/cal.com/tree/main/packages/config
> 3. https://github.com/calcom/cal.com/tree/main/packages/eslint-plugin
> 4. https://github.com/documenso/documenso/tree/feat/refresh/packages/eslint-config
> 5. https://github.com/documenso/documenso/tree/feat/refresh/packages/tsconfig
> 
> I see @Dhoni77 has contributed in cal.com, if you can guide or correct us here then it would be a great help @Dhoni77
> 
> @bathini79 you can start with creating the first package `@firecamp/eslint`, if you have any questions or need help then feel free to ask anytime here or in Discord.

Hi @Nishchit14 In cal.com they have configured prettier and eslint in a single package and exported it and referenced it in other packages if required.
May be we could also create new packages like you have suggested above and reference it across all packages.

**Comment by Nishchit14** [comment]
That's the plan now, thank you @Dhoni77

**Comment by Nishchit14** [comment]
@bathini79 are you working on it?

**Comment by bathini79** [comment]
Yes I'm working on it . And do we want to include all the above configs in one repo or multiple repo . And I managed to get the files. Earlier it was not showing in my git diff

**Comment by bathini79** [comment]
Okay. I've raised an initial pr.

**Comment by Nishchit14** [comment]
That's great @bathini79, I'll review it by tonight.
